lib/agent.js

```diff
@@ -113,7 +109,7 @@ Agent.prototype.destroy = function () {
   // for tests that may use multiple Agent instances in a single test process.
   global[symbols.agentInitialized] = null
 
-  if (Error.stackTraceLimit === this._conf.stackTraceLimit) {
+  if (this._origStackTraceLimit && Error.stackTraceLimit !== this._origStackTraceLimit) {
```

Comment:
REVIEW NOTE: This is a guard against agent.destroy() being called before agent.start()